lib/puppet/provider/foreman_resource/rest_v3.rb

```diff
@@ -55,6 +55,10 @@ def generate_token
     OAuth::AccessToken.new(oauth_consumer)
   end
 
+  def fqdn
+    URI::parse(resource[:base_url]).host
```

Comment:
```suggestion
    @fqdn ||= URI::parse(resource[:base_url]).host
```

So you end up calling `URI::parse` only once

Reply:
(one day I will decide whether I like Ruby's "magic returns" or not…)

Reply:
Done.